IA refresh homepage hero

Pomax opened the pull request:
Closes https://github.com/mozilla/foundation.mozilla.org/issues/4834

This updates the homepage template to use the same `get_banner` that banner pages do, with effectively a one-liner in the HomePage model to yield the hero banner when that function gets called.

Pomax commented on network-api/networkapi/settings.py:
```diff
@@ -242,6 +242,8 @@
     'whitenoise.middleware.WhiteNoiseMiddleware',
     'django.middleware.gzip.GZipMiddleware',
     'networkapi.utility.middleware.TargetDomainRedirectMiddleware',
+    'networkapi.utility.middleware.HTMLCleanUpMiddleware' if DEBUG else None,
```
I don't want to add this to prod, as it'll slow things down, but a source prettifier is quite useful. I can spin this off as a separate PR if we prefer, @mmmavis @cadecairos @KalobTaulien

mmmavis replied:
I vote for keeping them separate!

Pomax commented on network-api/networkapi/wagtailpages/pagemodels/base.py:
```diff
@@ -459,6 +459,9 @@ class Homepage(FoundationMetadataPageMixin, Page):
     )
 
+    def get_banner(self):
+        return self.hero_image
```
allows `{% with banner=page.specific.get_banner %}` in templates to work.

Pomax commented on network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:
```diff
@@ -1,34 +1,41 @@
-        {% if page.intro %}
-        <p class="body-large mt-3 mb-0">{{ page.intro }}</p>
-        {% endif %}
+      <div class="cutout col-12 pt-4">
```
this PR is not for the cutout, but these changes are here mostly so I can see things.

mmmavis replied:
Let's remove changes made to the cutout section since this affects other pages as well (e.g., hero section of primary page is broken due to this). Without the help of Percy it's hard for us to capture UI changes. It's better that we keep hero and cutout updates in separate PRs.

Pomax commented on source/sass/global.scss:
```diff
@@ -38,6 +38,21 @@ hr {
     overflow: hidden;
 
+    &.homepage {
+      $bg-pattern-height: 400px;
```
@mmmavis this is really more of an SCSS stub than anything, to get a rough idea before we actually write the real SCSS =)

mmmavis commented on network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:
```diff
@@ -1,34 +1,41 @@
-    <img src="{% static "_images/banner-black-white-marble.jpg" %}" alt="">
+  <div class="banner{% if homepage %} homepage{% endif %}">
+    {% if homepage %}
+      <div class="content dark-theme">
```
(Not sure if front end code needs review in this PR... is the code just for placeholder purpose only for now?)

If we want the layout to reflect more like the mockup in this PR - we should add the class `container` here.

Pomax replied:
all front-end work in this PR is purely "stub" work, it's not meant to be even remotely correct, just there to make sure there is a place to pick up from. I'll take it out of this PR if we want to do this once all the backend changes have landed.

mmmavis commented on source/sass/global.scss:
```diff
@@ -38,6 +38,22 @@ hr {
+        z-index: 1;
+      }
+
+      .cutout-wrapper-row {
```
Let's revert these changes too.

Pomax commented:
@mavis I've split off the html-formatting into a separate PR, and removed the template/css tweaks that I added to be able to see things. That said, I kept a bit in the primary heroguts fragment as placeholder for frontend work.
